feat(Narayana): update documentation to reference Narayana (Arjuna) (#686)
bonitasoft/bonita-doc: md/release-notes.md

## Technology updates

### JTA transaction manager replacement

In Bonita 7.9.0, we replaced the JTA transaction manager used to handle XA transactions in Bonita Engine from
Bitronix to Narayana (also known as Arjuna).  
This change should not impact the way to use Bonita.  

> However, tuning Bonita transaction configuration is a little different. If you are used to change default transaction timeout,

Reviewer: ```suggestion
However, tuning Bonita transaction configuration is now a little different. If you wish to change the default transaction timeout,
```

> it can now be done by changing the `defaultTimeout` property in file `server/conf/jbossts-properties.xml`  instead of

Reviewer: ```suggestion
it is now done by changing the `defaultTimeout` property in file `server/conf/jbossts-properties.xml`  instead of
```
